File: codeforce_personal/A_Consecutive_Sum.py
# have 2 max heap for even and odd
# from collections import defaultdict
# import heapq
for _ in range(int(input())):
    l,k = map(int, input().split())
    nums  = list(map(int,input().split()))
    ans = [0]*k

    for i in range(l):
        ans[i % k] = max(ans[i  % k], nums[i])
    print(sum(ans))    
    

File: codeforce_personal/A_Consecutive_Sum.cpp


void solve()
{
    long long n, k;
    cin >> n >> k;
    vector<long long> a(n), ans(k);

    for (int i = 0; i < n; i++)
        cin >> a[i];

    for (int i = 0; i < n; i++)
    {
        ans[(i + 1) % k] = max(ans[(i + 1) % k], a[i]);
    }

    long long out = 0;

    for (int i = 0; i < k; i++)
        out += ans[i];

    cout << out << endl;
}

signed main()
{
    int tc = 1;
    cin >> tc;
    for (int i = 1; i <= tc; i++)
    {
        solve();
    }
}